recipes/optimum/meta.yaml

```diff
@@ -0,0 +1,78 @@
+{% set name = "optimum" %}
+{% set version = "1.0.0" %}
+
+# Set "use_entry_points = True" once the entry-points 
+#    are properly defined in upstream.
+#    - Track Issue: https://github.com/huggingface/optimum/issues/95
+{% set use_entry_points = True %}
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/optimum-{{ version }}.tar.gz
```

Comment:
You probably need to change this to GitHub releases as the files might be slightly different here. That would be my first try.

Reply:
Since they have already made the change in the repository, this will be pushed to the next release on PyPI. So, if it's okay with you, I would, for now, prefer to stick to using the dynamic jinja2 alternative I had. 

Once they push the next PyPI release, all the extra stuff will be removed from the recipe in future.

Reply:
### Current `setup.py`

The change should start from line `#52` instead of `#65` (as in the patch for PR-96 in upstream). There were two other places where the code was different as well. So, I modified the patch to match these differences, and gave it a try with an adapted-patch.

<details>
<summary>Details</summary>
<p>

```python
import re

from setuptools import find_namespace_packages, setup


# Ensure we match the version set in src/optimum/version.py
try:
    filepath = "optimum/version.py"
    with open(filepath) as version_file:
        (__version__,) = re.findall('__version__ = "(.*)"', version_file.read())
except Exception as error:
    assert False, "Error: Could not open '%s' due %s\n" % (filepath, error)


install_requires = [
    "coloredlogs",
    "sympy",
    "transformers>=4.15.0",
    "torch>=1.9",
]

extras = {
    "onnxruntime": ["onnx", "onnxruntime", "datasets>=1.2.1"],
    "intel": ["pycocotools", "neural_compressor>=1.9", "datasets>=1.2.1", "pandas<1.4.0"],
    "graphcore": "optimum-graphcore",
}

setup(
    name="optimum",
    version=__version__,
    description="Optimum Library is an extension of the Hugging Face Transformers library, providing a framework to "
    "integrate third-party libraries from Hardware Partners and interface with their specific "
    "functionality.",
    long_description=open("README.md", "r", encoding="utf-8").read(),
    long_description_content_type="text/markdown",
    classifiers=[
        "Development Status :: 3 - Alpha",
        "License :: OSI Approved :: Apache Software License",
        "Intended Audience :: Developers",
        "Intended Audience :: Education",
        "Intended Audience :: Science/Research",
        "Operating System :: OS Independent",
        "Programming Language :: Python :: 3.8",
        "Programming Language :: Python :: 3.9",
        "Topic :: Scientific/Engineering :: Artificial Intelligence",
    ],
    keywords="transformers, quantization, pruning, training, ipu",
    url="https://huggingface.co/hardware",
    author="HuggingFace Inc. Special Ops Team",
    author_email="hardware@huggingface.co",
    license="Apache",
    packages=find_namespace_packages(include=["optimum*"]),
    install_requires=install_requires,
    extras_require=extras,
    entry_points={
        "console_scripts": [
            "optimum_export=optimum.onnxruntime.convert:main",
            "optimum_optimize=optimum.onnxruntime.optimize_model:main",
            "optimum_export_optimize=optimum.onnxruntime.convert_and_optimize:main",
        ],
    },
    include_package_data=True,
    zip_safe=False,
)
```

</p>
</details>